### PR TITLE
SECURESIGN-1461: check if Trillian's secret with db connection already exists

### PR DESCRIPTION
**Pull request by osmman**
fixes [SECURESIGN-1461](https://issues.redhat.com//browse/SECURESIGN-1461)
depends on https://github.com/securesign/secure-sign-operator/pull/673

### CODE REVIEW

File: internal/controller/trillian/actions/db/handle_secret.go [hunk 1]
```diff
@@ -108,3 +183,12 @@ func (i handleSecretAction) createDbSecret(namespace string, labels map[string]s
 		},
 	}
 }
+
+func (i handleSecretAction) secretAnnotations() map[string]string {
+	return map[string]string{
+		constants.LabelNamespace + "/mysql-database": "trillian",
```
Comment (bouskaJ) [inline internal/controller/trillian/actions/db/handle_secret.go:189]: Use constant (the hardcoded string is used also on L179)

File: internal/controller/trillian/actions/db/handle_secret.go [hunk 2]
```diff
@@ -108,3 +183,12 @@ func (i handleSecretAction) createDbSecret(namespace string, labels map[string]s
...
+		constants.LabelNamespace + "/mysql-user":     "mysql",
```
Comment (bouskaJ) [inline internal/controller/trillian/actions/db/handle_secret.go:190]: Use constant (the hardcoded string is used also on L180)

File: internal/controller/trillian/actions/db/handle_secret.go [hunk 3]
```diff
@@ -36,27 +51,94 @@ func (i handleSecretAction) Name() string {
...
 	if !utils.OptionalBool(instance.Spec.Db.Create) {
-		if instance.Spec.Db.DatabaseSecretRef != nil {
+		if instance.Spec.Db.DatabaseSecretRef == nil {
+			meta.SetStatusCondition(&instance.Status.Conditions, metav1.Condition{
+				Type:    trillian.DbCondition,
+				Status:  metav1.ConditionFalse,
+				Reason:  constants.Failure,
+				Message: "Expecting external DB configuration",
+			})
+			return i.StatusUpdate(ctx, instance)
```
Comment (bouskaJ) [inline internal/controller/trillian/actions/db/handle_secret.go:75]:
```suggestion
			return i.FailWithStatusUpdate(ctx, instance)
```

File: internal/controller/trillian/actions/db/handle_secret_test.go [hunk 4]
```diff
@@ -0,0 +1,672 @@
...
+			want: want{
+				result: testAction.StatusUpdate(),
+				verify: func(g Gomega, cli client.WithWatch, events <-chan watch.Event) {
+					instance := &rhtasv1alpha1.Trillian{}
+					g.Expect(cli.Get(context.TODO(), namespacedName, instance)).To(Succeed())
+
+					condition := meta.FindStatusCondition(instance.GetConditions(), actions2.DbCondition)
+					g.Expect(condition.Status).Should(Equal(metav1.ConditionFalse))
+					g.Expect(condition.Reason).Should(Equal(constants.Pending))
+
```
Comment (bouskaJ) [inline internal/controller/trillian/actions/db/handle_secret_test.go:529]: There is gomega support for channels. You can use something similar to https://github.com/securesign/secure-sign-operator/blob/080c130490bc5c5f84cf496721eb3ff150e9a265/internal/controller/fulcio/actions/servrConfig_test.go#L235-L238

Comment (osmman) [reply internal/controller/trillian/actions/db/handle_secret_test.go:529]: I tried but I unable to implement that logic by Gomega

File: internal/controller/trillian/actions/db/handle_secret.go (hunk 3 above)
Comment (osmman) [reply internal/controller/trillian/actions/db/handle_secret.go]: Done

**Pull request merged**